openstf/model/objective.py

```diff
@@ -31,19 +33,83 @@ class RegressorObjective:
         study.optimize(objective)
     """
 
+    def process_tuple(
+        self,
+        trial: optuna.trial.FrozenTrial,
+        key: str,
+        value: Tuple[Union[int, float], Union[int, float]],
+    ) -> Union[int, float]:
+        """Pick hyperparameter for tuple"""
+        tup, log = value
+        start_value, end_value = tup
+
+        error = "input values for the tuple can be float/float, float/int, int/float, int/int"
+
+        if isinstance(start_value, float) or isinstance(end_value, float):
+            return trial.suggest_float(
+                key, float(start_value), float(end_value), log=log
+            )
+
+        elif isinstance(start_value, int) and isinstance(end_value, int):
+            return trial.suggest_int(key, start_value, end_value)
+
+        else:
+            raise TypeError(error)
+
+    def get_trial_parameters(
+        self, model_params: dict, trial: optuna.trial.FrozenTrial
+    ) -> dict:
+        """
+        Pick hyperparameters from the hyperparameter space using optuna.
+        The dictionary can contain 2 possible types: tuple or List
+        1. a list can directly be used in the suggest_categorical
+        2. For the tuple we need a check to see what the tuple contains
+            a. two strings; it will pick one of the values based on the model used
+            b. one float with one integer or two floats;
+            will set the first value as lowerbound and the second value as upperbound and picks floats between those bounds.
+            This uses the log boolean which is defined in the tuple
+            c. two integers;
+            will set the first value as lowerbound and the second value as upperbound and picks integers between those bounds.
+            log boolean is ignored.
+
+        """
+        param = {}
+        for key, value in model_params.items():
+            # In the default parameter space we make use of tuples ((startValue: Union[int, float], endValue: Union[int, float]),log: bool)
+            if isinstance(value, tuple):
+                param[key] = self.process_tuple(trial, key, value)
+            # If the parameter is a list it means it is a categorical feature
+            elif isinstance(value, list):
+                param[key] = trial.suggest_categorical(key, value)
+
+            else:
+                raise TypeError("Possible input values: tuple, list")
+
+        return param
+
     def __init__(
         self,
         input_data,
+        model: OpenstfRegressorInterface,
+        pruning_function: optuna.integration,
+        model_params: dict,
```

Comment:
Is it required to use this argument? Does model.get_params() not return the same? If so, this interface is already a bit too large if we can make it smaller that would be better in my opinion.